crates/sequencer/src/sequencer.rs

```diff
@@ -629,54 +631,194 @@ where
+
+                    ledger_db
+                        .set_last_sequencer_commitment_l2_height(BatchNumber(
+                            commitment_info.l2_height_range.end().0,
+                        ))
+                        .map_err(|_| {
+                            anyhow!("Sequencer: Failed to set last sequencer commitment L2 height")
+                        })?;
+
+                    debug!("Commitment info: {:?}", commitment_info);
+
+                    let l2_start = l2_range_to_submit.start().0 as u32;
+                    let l2_end = l2_range_to_submit.end().0 as u32;
+                    if let Some(db_config) = db_config {
+                        match PostgresConnector::new(db_config).await {
+                            Ok(pg_connector) => {
+                                pg_connector
+                                    .insert_sequencer_commitment(
+                                        Into::<[u8; 32]>::into(tx_id).to_vec(),
+                                        l2_start,
+                                        l2_end,
+                                        commitment.merkle_root.to_vec(),
+                                        CommitmentStatus::Mempool,
+                                    )
+                                    .await
+                                    .map_err(|_| {
+                                        anyhow!("Sequencer: Failed to insert sequencer commitment")
+                                    })?;
+                            }
+                            Err(e) => {
+                                warn!("Failed to connect to postgres: {:?}", e);
+                            }
+                        }
                     }
+
+                    // Remove commitment from pending commitments
+                    let mut pending_commitments = pending_commitments_l2_range.lock().await;
+                    pending_commitments.retain(|&(start, end)| {
+                        start != *l2_range_to_submit.start() || end != *l2_range_to_submit.end()
+                    });
+                    ledger_db.set_pending_commitments_l2_range(&pending_commitments)?;
+
+                    info!("New commitment. L2 range: #{}-{}", l2_start, l2_end);
+                    Ok(())
                 }
-            }
+                .await;
 
-            // Clear state diff.
-            self.ledger_db.set_state_diff(vec![])?;
-            self.last_state_diff = vec![];
+                if let Err(err) = result {
+                    error!(
+                        "Error in spawned task for handling commitment result: {}",
+                        err
+                    );
+                }
+            });
+        }
+        Ok(())
+    }
 
-            info!("New commitment. L2 range: #{}-{}", l2_start, l2_end,);
+    #[instrument(level = "trace", skip(self), err, ret)]
+    pub async fn resubmit_pending_commitments(&mut self) -> anyhow::Result<()> {
```

Comment:
for simplicity, this function does not need to spawn a task

since it's called on restart, i think we can burden it taking a long time to start producing blocks. 

this is up for discussion tho.

Reply:
I would disagree with this suggestion. Because I can foresee that being really annoying.
1. developer restarts the app while there was a pending commitment to test a change
2. pending commitment takes like 5-10 minutes and blocks everything
3. developer angry

Reply:
We don't care much about the scenario you're proposing. The sequencer should not cater for the develop / test cycle compared to catering for consistency. We should wait for the pending commitments to finish before starting the sequencer normal operations. This could also prevent potential race conditions.

Reply:
> The sequencer should not cater for the develop / test cycle compared to catering for consistency.

I agree with this tbh.

But again @yaziciahmet has a good point on how it doesn't make a difference if we do this asynchronously.

however, I believe we should act defensively on these kind of issues.

since we don't expect this feature to be used frequently (well, hopefully never) it's a good idea to implement the simplest way possible.

let's go with the sync way.